src/managers/queue-manager.js

```diff
@@ -108,17 +70,21 @@ function handleMessage(message) {
 
 export async function pollMessageQueue() {
   if (getUserToken()) {
-    var response = await getUserQueue();
-    if (response && (response.status === 200 || response.status === 204)) {
-      log(`Message queue checked for user ${getUserToken()}, ${response.data.length} messages found.`);
-      if (response.data.length > 0) {
-        messages = response.data;
-        checkMessageQueue();
+    if (Gist.isDocumentVisible) {
+      var response = await getUserQueue();
+      if (response && (response.status === 200 || response.status === 204)) {
```

Comment:
Maybe abstract this into a util: 

isResponseSuccessful(response)?

Reply:
Since there are no other cases where this is being used, I think it's better we keep it like this.